src/component/segment_time.rs

```diff
@@ -0,0 +1,214 @@
+//! Provides the Segment Time Component. The Segment Time Component is a
+//! component that shows the time for the current segment in a comparison of
+//! your choosing. If no comparison is specified it uses the timer's current
+//! comparison.
+
+use super::key_value;
+use crate::comparison::best_segments;
+use crate::settings::{Color, Field, Gradient, SettingsDescription, Value};
+use crate::timing::formatter::{Accuracy, SegmentTime, TimeFormatter};
+use crate::{comparison, TimeSpan, Timer, TimerPhase, TimingMethod};
+use serde::{Deserialize, Serialize};
+use std::borrow::Cow;
+
+/// The Segment Time Component is a component that shows the time for the current
+/// segment in a comparison of your choosing. If no comparison is specified it
+/// uses the timer's current comparison.
+#[derive(Default, Clone)]
+pub struct Component {
+    settings: Settings,
+}
+
+/// The Settings for this component.
+#[derive(Clone, Serialize, Deserialize)]
+#[serde(default)]
+pub struct Settings {
+    /// The background shown behind the component.
+    pub background: Gradient,
+    /// The comparison chosen. Uses the Timer's current comparison if set to
+    /// `None`.
+    pub comparison_override: Option<String>,
+    /// Specifies whether to display the name of the component and its value in
+    /// two separate rows.
+    pub display_two_rows: bool,
+    /// The color of the label. If `None` is specified, the color is taken from
+    /// the layout.
+    pub label_color: Option<Color>,
+    /// The color of the value. If `None` is specified, the color is taken from
+    /// the layout.
+    pub value_color: Option<Color>,
+    /// The accuracy of the time shown.
+    pub accuracy: Accuracy,
+}
+
+impl Default for Settings {
+    fn default() -> Self {
+        Self {
+            background: key_value::DEFAULT_GRADIENT,
+            comparison_override: None,
+            display_two_rows: false,
+            label_color: None,
+            value_color: None,
+            accuracy: Accuracy::Hundredths, // TODO: Get this default from the formatter (some new constant).
```

Comment:
I'm also very unsure how you'd want me to go about doing this.
All of the other components have the accuracy specified directly rather than getting it from the formatter.